src/herder/test/HerderTests.cpp

```diff
@@ -2016,10 +2016,15 @@ TEST_CASE("herder externalizes values", "[herder]")
             if (i == ledgers.size() - 1)
             {
                 checkSynced(*(getC()));
+                // All the buffered ledgers are applied by now, so it's safe to
+                // trigger the next ledger
+                REQUIRE(herderC.mTriggerNextLedgerSeq == fourth + 1);
```

Comment:
this check is incomplete: what you mean here is that there is a "trigger timer" setup in the future and that it's "the right ledger" (same applies to other places where you check `mTriggerNextLedgerSeq` directly)

Reply:
I added a check for trigger timer setup. I left the section `older ledgers externalize and no-op` as-is, since it's supposed to skip triggering an older ledger, and the application may or may not have a trigger setup for the most recent ledger.